### PR TITLE
Track login error

### PR DESCRIPTION
**Pull request by juanlaube**
## What's this?
Tracks generic login error exceptions.

### Relationships

Closes nothing

## Why do we want this?
There's a possible bug out there in Daneel 1.14 and some users cannot login. This tracks the exception in a generic error.

## How is it done?
Tracking the exception.

### Side effects
None.

## :squid: Permissions
⛔️ Only me

### CODE REVIEW

File: Toggl.Foundation/Analytics/BaseAnalyticsService.cs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
```
Comment (simonrozsival) [inline Toggl.Foundation/Analytics/BaseAnalyticsService.cs]: 👀

File: Toggl.Foundation.MvvmCross/ViewModels/LoginViewModel.cs [hunk 2]
```diff
@@ -271,6 +271,8 @@ private void onError(Exception exception)
                     errorMessageSubject.OnNext("");
                     break;
                 default:
+                    analyticsService.UnknownLoginFailure.Track("Login failed with unknown reason", exception.StackTrace);
```
Comment (simonrozsival) [inline Toggl.Foundation.MvvmCross/ViewModels/LoginViewModel.cs]: Somehow include the `exception.Message` and `exception.GetType().FullName` so we get as much info as possible. The event name (`UnknownLoginFailure`) already says the same thing as `"Login failed with unknown reason"`.

File: Toggl.Foundation.MvvmCross/ViewModels/SignupViewModel.cs [hunk 3]
```diff
@@ -290,6 +290,8 @@ private void onError(Exception exception)
                     errorMessageSubject.OnNext(Resources.EmailIsAlreadyUsedError);
                     break;
                 default:
+                    analyticsService.UnknownLoginFailure.Track("Sign up failed with unknown reason", exception.StackTrace);
```
Comment (simonrozsival) [inline Toggl.Foundation.MvvmCross/ViewModels/SignupViewModel.cs]: Same as for login.

File: Toggl.Foundation/Analytics/BaseAnalyticsService.cs (hunk 1 above)
Comment (juanlaube) [reply Toggl.Foundation/Analytics/BaseAnalyticsService.cs]: 🤦‍♂️

**Pull request merged**